compiler/rustc_middle/src/mir/interpret/allocation.rs

```diff
@@ -479,10 +491,20 @@ impl<Prov: Provenance, Extra, Bytes: AllocBytes> Allocation<Prov, Extra, Bytes>
         self.provenance.clear(range, cx)?;
 
         assert!(range.end().bytes_usize() <= self.bytes.len()); // need to do our own bounds-check
+        // FIXME: actually now that `self.bytes` is a `Box`, this *does* invalidate existing
+        // immutable aliases at least under Stacked Borrows...
```

Comment:
I also noticed a UB bug in the interpreter itself here.^^
Maybe we should change this back to `Vec` for now? `Box` just doesn't have the APIs needed for this I think...